File: micro-service-gateway-server/src/main/java/demo/GatewayApplication.java
package demo;

import org.springframework.beans.factory.InitializingBean;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.gateway.route.RouteDefinitionRepository;
import org.springframework.context.annotation.Bean;
import org.springframework.core.annotation.Order;

@SpringBootApplication
public class GatewayApplication {
    @Autowired
    RouteDefinitionRepository routeDefinitionRepository;

    public static void main(String[] args) {
        SpringApplication.run(GatewayApplication.class, args);
    }


    @Bean
    @Order
    public InitializingBean init(MyConfig config) {

        return () -> {
            System.out.println(config.getMyName());
            routeDefinitionRepository.getRouteDefinitions()
                .subscribe(it -> {
                    System.out.println(it);
                });
        };
    }
}

File: micro-service-gateway-server/src/main/java/demo/GatewayApplication.kt
package demo

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cloud.gateway.route.RouteDefinitionRepository
import java.util.logging.*

@SpringBootApplication
class GatewayApplication : CommandLineRunner {
  @Autowired
  private lateinit var routeDefinitionRepository: RouteDefinitionRepository

  @Autowired
  private lateinit var config: MyConfig

  override fun run(vararg args: String?) {
    Logger.getAnonymousLogger().info { "GatewayApplication" + args.joinToString() }
  }
}

fun main(args: Array<String>) {
  runApplication<GatewayApplication>(*args)
}
